lambdas/service/app.py

```diff
@@ -153,10 +153,10 @@ def repository_search(entity_type: str, item_id: str):
         pagination = _get_pagination(app.current_request, entity_type)
         service = RepositoryService()
         return service.get_data(entity_type, pagination, filters, item_id, file_url)
-    except BadArgumentException as e:
-        raise BadRequestError(msg=e.message)
-    except IndexNotFoundError as e:
-        raise NotFoundError(msg=e.message)
+    except (BadArgumentException, InvalidUUIDError) as e:
+        raise BadRequestError(msg=e)
```

Comment:
This is relying on an implementation detail, specifically that BadRequestError's constructor implicitly applies `str()` to the argument.

```suggestion
        raise BadRequestError(msg=str(e))
```